tests/unit/test_task_storage.py

```diff
@@ -0,0 +1,191 @@
+    def __init__(self):
+        self.task_id = str(uuid.uuid1())
+        self.endpoint = None
+        self.status = TaskState.RECEIVED
+        self.result = None
+        self.result_reference = None
+
+
+def test_memory_storage_simple():
+    memstore = MemoryTaskStorage()
+    result = "foo result"
+    task = SimpleInMemoryTask()
+
+    b = memstore.store_result(task, result)
+    assert b is not None
+    assert memstore.get_result(task) == result
+
+
+def test_chained_storage_first_success():
+    memstore1 = MemoryTaskStorage()
+    memstore2 = MemoryTaskStorage()
+
+    chain = ChainedTaskStorage(memstore1, memstore2)
+
+    result = "foo result"
+    task = SimpleInMemoryTask()
+
+    b = chain.store_result(task, result)
+    assert b is True
+    assert memstore2.get_result(task) is None
+    assert memstore1.get_result(task) == result
+    assert chain.get_result(task) == result
+
+
+# @pytest.mark.skip
+def test_thresholded_storage_limit():
+    store = ThresholdedMemoryTaskStorage(result_limit_chars=10)
+
+    result = "result"
+    task = SimpleInMemoryTask()
+
+    b = store.store_result(task, result)
+    assert b is True
+    assert store.get_result(task) == result
+
+    result2 = "result too long for char limit"
+    task2 = SimpleInMemoryTask()
+
+    with pytest.raises(StorageException):
+        b = store.store_result(task2, result2)
+    with pytest.raises(StorageException):
+        store.get_result(task2)
+
+
+def test_chained_with_threshold():
+    memstore1 = ThresholdedMemoryTaskStorage(result_limit_chars=3)
+    memstore2 = MemoryTaskStorage()
+
+    chain = ChainedTaskStorage(memstore1, memstore2)
+
+    result = "foo result"
+    task = SimpleInMemoryTask()
+
+    b = chain.store_result(task, result)
+    assert b is True
+    with pytest.raises(StorageException):
+        memstore1.get_result(task)
+    assert memstore2.get_result(task) == result
+    assert chain.get_result(task) == result
+
+
+def test_null_storage():
+    store = NullTaskStorage()
+
+    result = "result"
+    task = SimpleInMemoryTask()
+
+    with pytest.raises(StorageException):
+        store.store_result(task, result)
+
+    with pytest.raises(StorageException):
+        store.get_result(task)
+
+
+def test_failing_chain_storage():
+    store1 = NullTaskStorage()
+    store2 = NullTaskStorage()
+
+    chain = ChainedTaskStorage(store1, store2)
+
+    result = "result"
+    task = SimpleInMemoryTask()
+
+    with pytest.raises(StorageException):
+        chain.store_result(task, result)
+    # maybe_todo : Figure out what the behavior here should be
+    # if the storage fails, should it be possible to retrieve
+    # the result?
+    # with pytest.raises(StorageException):
+    #    chain.get_result(task)
+
+
+@pytest.mark.skipif(
+    os.environ.get("AWS_PROFILE") is None,
+    reason="Test requires AWS creds until mock is added",
+)
+def test_s3_task_storage():
+    store = S3TaskStorage("funcx-test-1")
```

Comment:
The way this is written, I can't run the tests in my AWS account because I can't `aws s3 mb s3://funcx-test-1` (because that bucket already exists).

Can we make this an env var which both controls the test skips (instead of `AWS_PROFILE`) and sets the bucket name to use for testing? That would allow others to update `tox.ini` to pass the env var and run `FUNCX_COMMON_TEST_S3_BUCKET=foo tox` to see accurate coverage data.

Reply:
I just pushed an update that uses moto to mock out the unit tests.  That should satisfy both CI and provide code coverage locally.   If we want to eventually support testing against a real AWS instance by configuring tests on a per user basis, we can address that later.